### PR TITLE
OF and LR updates

### PR DESCRIPTION
**Pull request by BroccoliHijinx**
Taking into account review notes.  Having trouble setting up my python environment, so I have not been able to test the code yet.  I hope to fix that today/tomorrow.  Fixed the table in LR.

### CODE REVIEW

File: code/overview/overfitting/example.py
Comment (hokie45) [inline code/overview/overfitting/example.py]: Yeah This code doesn't work, also you should explain a little more in comments what the code does. Also change file name to something more descriptive

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (hokie45) [inline code/supervised/Logistic_Regression/logistic_ex1.py]: Capital 'T' in true

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (hokie45) [inline code/supervised/Logistic_Regression/logistic_ex1.py]: explain what X and y are

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (hokie45) [inline code/supervised/Logistic_Regression/logistic_ex1.py]: I'm not sure why you are predicting things you trained the model on. I would suggest creating some sample vectors [5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]
like this but change the values because that is part of the training set

File: docs/source/content/overview/overfitting.rst
Comment (hokie45) [inline docs/source/content/overview/overfitting.rst:20]: Add a conclusion and probably a motivation section

Comment (hokie45) [inline docs/source/content/overview/overfitting.rst:20]: Also also add links to code, and have code snippets to show how it relates to overfitting

File: docs/source/content/supervised/logistic_regression.rst
Comment (hokie45) [inline docs/source/content/supervised/logistic_regression.rst]: Add motivation section and add code snippets and show how it relates to the doc.

**Review by hokie45 — CHANGES_REQUESTED** [review]

**Review by mergu — COMMENTED** [review]
Please also squash these commits together

**Comment by BroccoliHijinx** [comment]
Logistic example is functional, please take a look and request any further changes.

**Review by mergu — CHANGES_REQUESTED** [review]
Please also squash these commits together into more meaningful changes

**Comment by BroccoliHijinx** [comment]
Working on squashing, but having trouble after having already merged master in.  In the future, I will be sure to do a rebase and squash commits.  For now I will keep looking into it, sorry it's such a mess.

The main changes since last review are as follows (in case I cannot squash successfully):

example.py for overfitting completely redone
logistic_ex1.py completely redone
Overfitting has had a motivation section added
Logistic Regression has had a motivation and conclusion section added.

**Comment by BroccoliHijinx** [comment]
Okay, I have squashed the commits so they should be more readable

File: code/overview/overfitting/example.py
Comment (mergu) [inline code/overview/overfitting/example.py]: Can you use matplotlib to output these graphs too? I think it'd be beneficial for the user to see a visual example

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (mergu) [inline code/supervised/Logistic_Regression/logistic_ex1.py]: this could be combined into the above expression with a ternary statement instead of re-looping over 1000 elements

File: docs/source/content/overview/overfitting.rst
Comment (mergu) [inline docs/source/content/overview/overfitting.rst]: this image was not created from that desmos site

File: docs/source/content/overview/overfitting.rst
Comment (mergu) [inline docs/source/content/overview/overfitting.rst]: please add a section explaining the code implementation (also with a link to the code)

File: docs/source/content/supervised/logistic_regression.rst
Comment (mergu) [inline docs/source/content/supervised/logistic_regression.rst]: please add a section explaining the code implementation (also with a link to the code)

**Review by mergu — CHANGES_REQUESTED** [review]
In terms of overall formatting, I have a few suggestions:

The first thing I recommend is to change all of the double spaces after periods to single spaces. GitHub doesn't display double spaces and I'm not sure whether the site will show double spaces, so for consistency everything should be single spaced. Also, some of your lines are too long so it makes it difficult to comment on any specific points. Take a look at existing merged docs for general line lengths

My second point is I think the sections need to have more content. This was discussed in a couple client meetings I think you missed.

File: code/overview/overfitting/example.py
Comment (BroccoliHijinx) [reply code/overview/overfitting/example.py]: matplotlib graph added in latest commit

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (BroccoliHijinx) [reply code/supervised/Logistic_Regression/logistic_ex1.py]: I agree, it is more efficient, but I kept it like this so that line doesn't become too difficult to follow.

File: docs/source/content/overview/overfitting.rst
Comment (BroccoliHijinx) [reply docs/source/content/overview/overfitting.rst]: Correct, I swapped it with one of the Wolfram captions by accident.

**Comment by BroccoliHijinx** [comment]
See f1fce12 for change list.  Formatting should be good now, and addressed all comments.  Added a multinomial LR section.

File: docs/source/content/overview/overfitting.rst
Comment (mergu) [inline docs/source/content/overview/overfitting.rst]: regularization typo

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (hokie45) [inline code/supervised/Logistic_Regression/logistic_ex1.py:1]: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning. 
I got this warning from this code, I'm not sure what it means, was wondering if you knew?

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (hokie45) [inline code/supervised/Logistic_Regression/logistic_ex1.py]: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True).

I got this warning also.

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (hokie45) [inline code/supervised/Logistic_Regression/logistic_ex1.py:46]: Explain a little on what the output means,  
for example, 
[673]: Class 0, probability [0.59449694 0.40550306]
explain the two probability numbers and the [673]: 


also i would put a new line before you print out 100 out of 100 correct, just looks a little nicer.

**Review by hokie45 — CHANGES_REQUESTED** [review]

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (BroccoliHijinx) [reply code/supervised/Logistic_Regression/logistic_ex1.py:1]: Yeah, the LogisticRegression objects can use multiple different 'solvers' (basically just methods of finding the solution), which would be specified in the object creation.  I'll add that specifier and the warning should go away.  I think we are running on different versions of sklearn.

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (BroccoliHijinx) [reply code/supervised/Logistic_Regression/logistic_ex1.py]: I think this is also from different versions, but I might be wrong.  I'll look into what is the latest practice.

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (BroccoliHijinx) [reply code/supervised/Logistic_Regression/logistic_ex1.py:46]: Left a comment with the explanation and an example, and added a new line

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (BroccoliHijinx) [reply code/supervised/Logistic_Regression/logistic_ex1.py:1]: I never saw the warning, but it should be gone with the latest version.  Please verify this when you get a chance.  All that should have been necessary was the specifier in the object.

File: docs/source/content/overview/overfitting.rst
Comment (BroccoliHijinx) [reply docs/source/content/overview/overfitting.rst]: Fixed in c1235da

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (BroccoliHijinx) [reply code/supervised/Logistic_Regression/logistic_ex1.py]: Switched to a simple numpy array, please verify the warning goes away when you get the chance.

File: docs/source/content/supervised/logistic_regression.rst
Comment (mergu) [inline docs/source/content/supervised/logistic_regression.rst:94]: This link is never defined so it just leads back to this page instead of the code

File: docs/source/content/supervised/logistic_regression.rst
Comment (mergu) [inline docs/source/content/supervised/logistic_regression.rst:113]: You need a newline before this link for it to work properly

**Comment by astorfi** [comment]
@weric16 @mergu @hokie45 Thank you all very much. Please accelerate the process if possible.

**Comment by astorfi** [comment]
@weric16 Once the reviewers approved your changes, please make the whole thing as one consistent pull request. Thanks

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (BroccoliHijinx) [reply code/supervised/Logistic_Regression/logistic_ex1.py]: Turned it into a function in the end.

File: docs/source/content/supervised/logistic_regression.rst
Comment (BroccoliHijinx) [reply docs/source/content/supervised/logistic_regression.rst:113]: Fixed in a7fc1f2

File: docs/source/content/supervised/logistic_regression.rst
Comment (BroccoliHijinx) [reply docs/source/content/supervised/logistic_regression.rst:94]: Should be fixed in a7fc1f2

File: docs/source/content/overview/overfitting.rst
Comment (mergu) [inline docs/source/content/overview/overfitting.rst]: This appears as a quote since it has a space at the front

File: docs/source/content/supervised/logistic_regression.rst
Comment (mergu) [inline docs/source/content/supervised/logistic_regression.rst:9]: Can you add a table of contents at the top of this like the overfitting section?

Comment (BroccoliHijinx) [reply docs/source/content/supervised/logistic_regression.rst:9]: 9185ec3

File: docs/source/content/overview/overfitting.rst
Comment (BroccoliHijinx) [reply docs/source/content/overview/overfitting.rst]: 9185ec3

**Review by mergu — APPROVED** [review]
Approving the docs portion of this PR, I'll let Zac comment/approve the code

**Comment by astorfi** [comment]
@hokie45 Would you please comment here if you approve or offering more changes?

File: code/supervised/Logistic_Regression/logistic_ex1.py
Comment (hokie45) [reply code/supervised/Logistic_Regression/logistic_ex1.py:1]: Yeah the warning is gone, thanks the comments look good.

**Review by hokie45 — APPROVED** [review]

**Comment by astorfi** [comment]
@weric16 Please create one commit with all the changes ... Thanks

**Comment by BroccoliHijinx** [comment]
I squashed all the commits into one @astorfi

**Comment by astorfi** [comment]
@weric16 Thank you so much

**Comment by astorfi** [comment]
@mergu @hokie45 Thank you for reviewing.

**Pull request merged**
